ui/lib/core/addon/components/search-select.js

```diff
@@ -172,7 +177,26 @@ export default Component.extend({
     return filterOptions(options || [], searchText, matcher);
   },
   // -----
-
+  customizeObject(option) {
+    // if passObject=true return array of objects
+    if (this.passObject) {
+      let additionalKeys;
+      if (this.objectKeys) {
+        // if the LIST endpoint has key_info and the model has been hydrated in serializer (ex: serializer/oidc/clients)
+        // 'option' is a model record, so pull any attrs we want sent to the parent by adding them to the selected option (object)
+        additionalKeys = Object.fromEntries(this.objectKeys.map((key) => [key, option[key]]));
+        if (Object.values(additionalKeys).includes(undefined)) additionalKeys = false;
```

Comment:
if `additionalKeys` contains `undefined` values then the option is new, so we set to false because if `additionalKeys=false` it won't spread and be added into the object below on line 195.

(open to better ways of handling this, as it feels a bit wonky)

Reply:
Could we instead filter after it's mapped? As I read this, if *any* of the `additionalKeys` have an undefined value, none of the requested additional keys are returned

Reply:
That was originally the functionality I was going for - but a good point to return any keys that _do_ have value.